tokio-util/src/sync/tests/loom_cancellation_token.rs

```diff
@@ -174,3 +174,46 @@ fn cancel_parent_and_child() {
         assert_ok!(th3.join());
     });
 }
+
+#[test]
+fn run_until_cancelled_completes() {
+    loom::model(|| {
+        block_on(async {
+            let token = CancellationToken::new();
```

Comment:
You are not using loom tests correctly. But you also do not need a loom test for this. You can add a normal test in the `tokio-util/tests` directory instead of `tokio-util/src/sync/tests`.

Reply:
Should be done.